compiler/testData/diagnostics/tests/suspendConversion/overloadResolutionBySuspendModifier.fir.kt

```diff
@@ -14,6 +14,6 @@ fun test1() {
 
 // candidate without suspend conversions is more specific
 fun test2(f: () -> Int, g: suspend () -> Int) {
-    foo(f)
+    <!AMBIGUITY!>foo<!>(f)
```

Comment:
It's a regression, we should not have ambiguity here

Reply:
This is what I meant about `useSuspendConversion` flag inside `Candidate`. As mentioned in the comment
```kt
// candidate without suspend conversions is more specific
```
we should distinguish candidates with and without suspend conversion.

Reply:
Are you planning to implement this a bit later?

Reply:
I locally implemented it, and am able to resolve those ambiguities. Will work on refactoring as well.

Reply:
Implemented at the 2nd commit: https://github.com/JetBrains/kotlin/pull/3710/commits/b17f7f51a59a85ef7fd80aec0541c066857a026c where all ambiguity regressions added by the 1st commit are resolved.